File: core/baker-annotations/src/main/java/com/ing/baker/recipe/annotations/AsyncInteraction.java
package com.ing.baker.recipe.annotations;

import javax.inject.Qualifier;
import java.lang.annotation.*;

/**
 * Annotation for an interaction to show it supports Async calls
 */
@Qualifier
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface AsyncInteraction {

}

File: core/baker-annotations/src/main/kotlin/com/ing/baker/recipe/annotations/AsyncInteraction.kt
package com.ing.baker.recipe.annotations

import javax.inject.Qualifier

/**
 * Annotation for an interaction to show it supports Async calls
 */
@Qualifier
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION)
annotation class AsyncInteraction
